.github/workflows/test.yml

```diff
@@ -1,69 +1,56 @@
-
-    env:
-      RUBYOPT: "-w"
-      ALL_WARNINGS: "/tmp/all-warnings"
-
     strategy:
-      fail-fast: false
       matrix:
         ruby-version: ["3.2", "3.1", "3.0", "2.7"]
-
     steps:
-    - uses: actions/checkout@v4
-
-    - name: Install OS dependencies
-      run: |
-        sudo apt-get update -qq
-        sudo apt-get install --assume-yes libcurl4-openssl-dev
-
-    - name: Set up Ruby ${{ matrix.ruby-version }}
-      uses: ruby/setup-ruby@v1
-      with:
-        ruby-version: ${{ matrix.ruby-version }}
-        bundler-cache: true
-
-    - name: faraday-1.0.0
-      run: |
-        bundle install
-        bundle exec rspec spec/lib/vcr/middleware/faraday_spec.rb spec/lib/vcr/library_hooks/faraday_spec.rb 2>> "${ALL_WARNINGS}"
-        bundle exec cucumber features/middleware/faraday.feature 2>> "${ALL_WARNINGS}"
-      env:
-        BUNDLE_GEMFILE: Gemfile.faraday-1.0.0
-
-    - name: cucumber
-      run: |
-        bundle install
-        bundle exec cucumber features/test_frameworks/cucumber.feature 2>> "${ALL_WARNINGS}"
-      env:
-        BUNDLE_GEMFILE: Gemfile.cucumber
-
-    - name: rspec
-      run: |
-        bundle install
-        bundle exec rspec spec/ 2>> "${ALL_WARNINGS}"
-
-    - name: cucumber
-      run: |
-        bundle install
-        bundle exec cucumber features/ 2>> "${ALL_WARNINGS}"
-
-    - name: check warnings
-      run: |
-        vcr_warnings="$(grep -F "$PWD" "${ALL_WARNINGS}" | grep "warning: " | grep -v "${PWD}/vendor/bundle" | sort | uniq -c | tee /dev/stderr | wc -l)"
-        if [ "$vcr_warnings" -gt 0 ]; then echo "FAILED: test suite doesn't tolerate warnings"; exit 1; fi
-
-    - name: doc coverage
-      run: |
-        bundle install
-        bundle exec yard stats --list-undoc | tee /dev/stdout | grep -q '100.00% documented'
+      - uses: actions/checkout@v4
+      - name: Install OS dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install --assume-yes libcurl4-openssl-dev
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: cucumber
+        run: ./script/fail_if_warnings cucumber features/
+  rspec:
+    name: "RSpec / ${{ matrix.ruby-version }}"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ["3.2", "3.1", "3.0", "2.7"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install OS dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install --assume-yes libcurl4-openssl-dev
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: rspec
+        run: ./script/fail_if_warnings rspec spec/
+  doc-coverage:
+    name: "Doc coverage"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "ruby"
```

Comment:
🗒️ I assumed that running the yard coverage check against a matrix of multiple versions of Ruby was overkill. I changed it to run on the latest version of Ruby only.